### ISSUE TITLE
Rasa 3 - form activation from custom action failing with several approaches

### ISSUE THREAD

**Issue by Hendler** [RasaHQ/rasa#11049]
### Rasa Open Source version

3.1.0

### Rasa SDK version

 _No response_ 

### Rasa X version

 _No response_ 

### Python version

3.8

### What operating system are you using?

OSX

### What happened?

Sorry if this is not a bug and is a user error. Simply put, I seem to not be able to trigger a form to work from a custom action, where the for may or may not not be triggered there.

The rule seems to work correctly when the intent is directly triggered, but other paths for triggering the form via UserUttered, conditional slotset, and other approaches have failed. I'm playing with the config policies now, but getting even worse results in terms of unknown next actions, fallbacks etc.

Have mention the repo, which has three branches I'm trying to resolve the issue in [https://github.com/Hendler/rasa-form-from-custom-action](https://github.com/Hendler/rasa-form-from-custom-action). I also posted in the forum, [and that may a better place](https://forum.rasa.com/t/problem-triggering-a-form-from-a-custom-action-in-rasa-3/52294)

If this is not a bug, I wonder if there can be some better detection of rules that won't work if there is missing info.

### Command / Request

<pre>The form is:

forms:
 my_form:
 ignored_intents:

*   chitchat
     required_slots:
*   my_slot
     </pre>

Slot is:

 my_slot:
 type: any
 mappings:

*   type: from_entity
     entity: my_slot

For `my_form_trigger`

 my_form_trigger:
 type: bool
 influence_conversation: true
 mappings:

*   type: custom

The rules looks like so:

*   rule: Activate My Form
     steps:
*   intent: activate_my_form
*   action: utter_form_start
*   action: my_form
*   active_loop: my_form
*   slot_was_set:
*   my_form_trigger

*   rule: Finished with My Form
     condition:
*   active_loop: my_form
     steps:

1.  Form is deactivated

*   action: my_form
*   active_loop: null
*   slot_was_set:
*   my_slot: null
*   action: utter_my_form_slots_values

config

1.  The config recipe.
2.  [https://rasa.com/docs/rasa/model-configuration/](https://rasa.com/docs/rasa/model-configuration/)
     recipe: default.v1

1.  Configuration for Rasa NLU.
2.  [https://rasa.com/docs/rasa/nlu/components/](https://rasa.com/docs/rasa/nlu/components/)
     language: en
     pipeline:

*   name: SpacyNLP
     model: "en_core_web_md"

1.  model: "en_core_web_lg"

*   name: SpacyTokenizer
*   name: SpacyEntityExtractor
*   name: SpacyFeaturizer
     pooling: mean
*   name: CountVectorsFeaturizer
     analyzer: char_wb
     min_ngram: 1
     max_ngram: 4
*   name: DIETClassifier
     epochs: 100
     constrain_similarities: true

1.  model_confidence: cosine
2.  model_confidence: linear_norm
     model_confidence: softmax

*   name: EntitySynonymMapper
*   name: ResponseSelector
     epochs: 64
     constrain_similarities: true

1.  - name: FallbackClassifier
2.  threshold: 0.3
3.  ambiguity_threshold: 0.1

*   name: LanguageModelFeaturizer

1.  Name of the language model to use
     model_name: "bert"
2.  Pre-Trained weights to be loaded
     model_weights: "rasa/LaBSE"
3.  An optional path to a specific directory to download and cache the pre-trained model weights.
4.  The `default` cache_dir is the same as [https://huggingface.co/transformers/serialization.html#cache-directory](https://huggingface.co/transformers/serialization.html#cache-directory) .
     cache_dir: null

1.  # No configuration for the NLU pipeline was provided. The following default pipeline was used to train your model.
2.  # If you'd like to customize it, uncomment and adjust the pipeline.
3.  # See [https://rasa.com/docs/rasa/tuning-your-model](https://rasa.com/docs/rasa/tuning-your-model) for more information.
4.  - name: WhitespaceTokenizer
5.  - name: RegexFeaturizer
6.  - name: LexicalSyntacticFeaturizer
7.  - name: CountVectorsFeaturizer
8.  - name: CountVectorsFeaturizer
9.  analyzer: char_wb
10.  min_ngram: 1
11.  max_ngram: 4
12.  - name: DIETClassifier
13.  epochs: 100
14.  - name: EntitySynonymMapper
15.  - name: ResponseSelector
16.  epochs: 100
17.  - name: FallbackClassifier
18.  threshold: 0.3
19.  ambiguity_threshold: 0.1

1.  Configuration for Rasa Core.
2.  [https://rasa.com/docs/rasa/core/policies/](https://rasa.com/docs/rasa/core/policies/)
     policies:

*   name: MemoizationPolicy

1.  TEDPolicy max_history: 5 epochs: 100 constrain_similarities: true

*   name: AugmentedMemoizationPolicy
*   name: RulePolicy

1.  TODO use this to hand back to human
2.  core_fallback_threshold: 0.4
3.  core_fallback_action_name: "action_default_fallback"
4.  enable_fallback_prediction: False
5.  - name: FormPolicy

1.  - name: UnexpecTEDIntentPolicy
2.  max_history: 5
3.  epochs: 64
4.  - name: TEDPolicy
5.  max_history: 5
6.  epochs: 64
7.  constrain_similarities: true
8.  model_confidence : softmax

1.  # No configuration for policies was provided. The following default policies were used to train your model.
2.  # If you'd like to customize them, uncomment and adjust the policies.
3.  # See [https://rasa.com/docs/rasa/policies](https://rasa.com/docs/rasa/policies) for more information.
4.  - name: MemoizationPolicy
5.  - name: TEDPolicy
6.  max_history: 5
7.  epochs: 100
8.  - name: RulePolicy

1.  1.  1.  Relevant log output

 ```shell
 2022-03-31 12:13:25 DEBUG rasa.core.lock_store - Issuing ticket for conversation '93ae2dfd7824465ab395b1d15ffe50bf'.
 2022-03-31 12:13:25 DEBUG rasa.core.lock_store - Acquiring lock for conversation '93ae2dfd7824465ab395b1d15ffe50bf'.
 2022-03-31 12:13:25 DEBUG rasa.core.lock_store - Acquired lock for conversation '93ae2dfd7824465ab395b1d15ffe50bf'.
 2022-03-31 12:13:25 DEBUG rasa.core.tracker_store - Recreating tracker for id '93ae2dfd7824465ab395b1d15ffe50bf'
 2022-03-31 12:13:25 DEBUG rasa.engine.runner.dask - Running graph with inputs: 

 {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x1afaef460>]} 

, targets: <span class="error">['run_RegexMessageHandler']</span> and ExecutionContext(model_id='f832a7635ad144a0b808a5b94e53e38f', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
 2022-03-31 12:13:25 DEBUG rasa.engine.graph - Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
 2022-03-31 12:13:25 DEBUG rasa.engine.graph - Node 'provide_SpacyNLP0' running 'SpacyNLP.provide'.
 2022-03-31 12:13:25 DEBUG rasa.engine.graph - Node 'run_SpacyNLP0' running 'SpacyNLP.process'.
 2022-03-31 12:13:25 DEBUG rasa.engine.graph - Node 'run_SpacyTokenizer1' running 'SpacyTokenizer.process'.
 2022-03-31 12:13:25 DEBUG rasa.engine.graph - Node 'run_SpacyEntityExtractor2' running 'SpacyEntityExtractor.process'.
 2022-03-31 12:13:25 DEBUG rasa.engine.graph - Node 'run_SpacyFeaturizer3' running 'SpacyFeaturizer.process'.
 2022-03-31 12:13:25 DEBUG rasa.engine.graph - Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_DIETClassifier5' running 'DIETClassifier.process'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_EntitySynonymMapper6' running 'EntitySynonymMapper.process'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_ResponseSelector7' running 'ResponseSelector.process'.
 2022-03-31 12:13:26 DEBUG rasa.nlu.classifiers.diet_classifier - There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
 2022-03-31 12:13:26 DEBUG rasa.nlu.selectors.response_selector - Adding following selector key to message property: default
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_FallbackClassifier8' running 'FallbackClassifier.process'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_LanguageModelFeaturizer9' running 'LanguageModelFeaturizer.process'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'domain_provider' running 'DomainProvider.provide_inference'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Received user message 'custom action intent' with intent '

 {'name': 'custom_action_intent', 'confidence': 0.9994083642959595} 

' and entities '[]'
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Logged UserUtterance - tracker now has 9 events.
 2022-03-31 12:13:26 DEBUG rasa.core.actions.action - Validating extracted slots:
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Default action 'action_extract_slots' was executed, resulting in 0 events:
 2022-03-31 12:13:26 DEBUG rasa.engine.runner.dask - Running graph with inputs: 

 {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x1afaef5e0>} 

, targets: <span class="error">['select_prediction']</span> and ExecutionContext(model_id='f832a7635ad144a0b808a5b94e53e38f', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'domain_provider' running 'DomainProvider.provide_inference'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.memoization - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 <span class="error">[state 3]</span> user intent: custom_action_intent | previous action name: action_listen
 2022-03-31 12:13:26 DEBUG rasa.core.policies.memoization - There is no memorised next action
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.rule_policy - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 <span class="error">[state 3]</span> user text: custom action intent | previous action name: action_listen
 2022-03-31 12:13:26 DEBUG rasa.core.policies.rule_policy - There is no applicable rule.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.rule_policy - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 <span class="error">[state 3]</span> user intent: custom_action_intent | previous action name: action_listen
 2022-03-31 12:13:26 DEBUG rasa.core.policies.rule_policy - There is no applicable rule.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.ted_policy - TED predicted 'utter_greet' based on user intent.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.unexpected_intent_policy - Querying for intent `custom_action_intent`.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.unexpected_intent_policy - Query intent index 4 not found in label thresholds - 

 {1: -0.38845077, 5: -0.7065364, 7: -0.026301295, 8: -0.24622284, 9: -0.40931982} 

. Check for `action_unlikely_intent` prediction will be skipped.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.ensemble - Made prediction using user intent.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.ensemble - Added `DefinePrevUserUtteredFeaturization(False)` event.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.ensemble - Predicted next action using RulePolicy.
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Predicted next action 'action_default_fallback' with confidence 0.30.
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Policy prediction ended with events '<span class="error">[
 <rasa.shared.core.events.defineprevuserutteredfeaturization object at 0x1afaea4f0>
  ]
 </rasa.shared.core.events.defineprevuserutteredfeaturization></span>'.
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Action 'action_default_fallback' ended with events '<span class="error">[
 <rasa.shared.core.events.userutterancereverted object at 0x1af876fd0>
  ]
 </rasa.shared.core.events.userutterancereverted></span>'.
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Current slot values:
 my_slot: None
 my_form_trigger: None
 requested_slot: None
 session_started_metadata: None
 2022-03-31 12:13:26 DEBUG rasa.engine.runner.dask - Running graph with inputs: 

 {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x1afaef5e0>} 

, targets: <span class="error">['select_prediction']</span> and ExecutionContext(model_id='f832a7635ad144a0b808a5b94e53e38f', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'domain_provider' running 'DomainProvider.provide_inference'.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.memoization - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 2022-03-31 12:13:26 DEBUG rasa.core.policies.memoization - There is a memorised next action 'action_listen'
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.rule_policy - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 2022-03-31 12:13:26 DEBUG rasa.core.policies.rule_policy - There is no applicable rule.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.ted_policy - TED predicted 'action_listen' based on user intent.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.unexpected_intent_policy - Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
 2022-03-31 12:13:26 DEBUG rasa.engine.graph - Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
 2022-03-31 12:13:26 DEBUG rasa.core.policies.ensemble - Predicted next action using MemoizationPolicy.
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Predicted next action 'action_listen' with confidence 1.00.
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Policy prediction ended with events '[]'.
 2022-03-31 12:13:26 DEBUG rasa.core.processor - Action 'action_listen' ended with events '[]'.
 2022-03-31 12:13:26 DEBUG rasa.core.lock_store - Deleted lock for conversation '93ae2dfd7824465ab395b1d15ffe50bf'.

 Activating intent also is an issue

 2022-03-31 12:22:31 DEBUG rasa.core.lock_store - Issuing ticket for conversation '61f6d7f4fa2f44e7834b71cf72e8d2b7'.
 2022-03-31 12:22:31 DEBUG rasa.core.lock_store - Acquiring lock for conversation '61f6d7f4fa2f44e7834b71cf72e8d2b7'.
 2022-03-31 12:22:31 DEBUG rasa.core.lock_store - Acquired lock for conversation '61f6d7f4fa2f44e7834b71cf72e8d2b7'.
 2022-03-31 12:22:31 DEBUG rasa.core.tracker_store - Recreating tracker for id '61f6d7f4fa2f44e7834b71cf72e8d2b7'
 2022-03-31 12:22:31 DEBUG rasa.engine.runner.dask - Running graph with inputs: 

 {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x1b1bcda30>]} 

, targets: <span class="error">['run_RegexMessageHandler']</span> and ExecutionContext(model_id='11ec92063cb6476b839b182307917603', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'provide_SpacyNLP0' running 'SpacyNLP.provide'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_SpacyNLP0' running 'SpacyNLP.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_SpacyTokenizer1' running 'SpacyTokenizer.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_SpacyEntityExtractor2' running 'SpacyEntityExtractor.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_SpacyFeaturizer3' running 'SpacyFeaturizer.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_DIETClassifier5' running 'DIETClassifier.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_EntitySynonymMapper6' running 'EntitySynonymMapper.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_ResponseSelector7' running 'ResponseSelector.process'.
 2022-03-31 12:22:31 DEBUG rasa.nlu.classifiers.diet_classifier - There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
 2022-03-31 12:22:31 DEBUG rasa.nlu.selectors.response_selector - Adding following selector key to message property: default
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_FallbackClassifier8' running 'FallbackClassifier.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_LanguageModelFeaturizer9' running 'LanguageModelFeaturizer.process'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'domain_provider' running 'DomainProvider.provide_inference'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Received user message 'hello' with intent '

 {'name': 'greet', 'confidence': 0.9999849796295166} 

' and entities '[]'
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Logged UserUtterance - tracker now has 20 events.
 2022-03-31 12:22:31 DEBUG rasa.core.actions.action - Validating extracted slots:
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Default action 'action_extract_slots' was executed, resulting in 0 events:
 2022-03-31 12:22:31 DEBUG rasa.engine.runner.dask - Running graph with inputs: 

 {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x1b1aed160>} 

, targets: <span class="error">['select_prediction']</span> and ExecutionContext(model_id='11ec92063cb6476b839b182307917603', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'domain_provider' running 'DomainProvider.provide_inference'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.memoization - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 <span class="error">[state 3]</span> user intent: greet | previous action name: action_listen
 2022-03-31 12:22:31 DEBUG rasa.core.policies.memoization - There is no memorised next action
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.rule_policy - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 <span class="error">[state 3]</span> user text: hello | previous action name: action_listen
 2022-03-31 12:22:31 DEBUG rasa.core.policies.rule_policy - There is no applicable rule.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.rule_policy - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 <span class="error">[state 3]</span> user intent: greet | previous action name: action_listen
 2022-03-31 12:22:31 DEBUG rasa.core.policies.rule_policy - There is no applicable rule.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.ted_policy - TED predicted 'utter_greet' based on user intent.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.unexpected_intent_policy - Querying for intent `greet`.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.unexpected_intent_policy - Score for intent `greet` is `-1.1194305419921875`, while threshold is `-0.21044859290122986`.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.unexpected_intent_policy - Top 5 intents (in ascending order) that are likely here are: `<span class="error">[('session_start', -0.5387745), ('deny', -0.45278916), ('affirm', -0.3587651), ('mood_great', 0.7020666), ('mood_unhappy', 0.7211871)]</span>`.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.unexpected_intent_policy - Intent `greet-7` unlikely to occur here.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.ensemble - Made prediction using user intent.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.ensemble - Added `DefinePrevUserUtteredFeaturization(False)` event.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.ensemble - Predicted next action using UnexpecTEDIntentPolicy.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Predicted next action 'action_unlikely_intent' with confidence 1.00.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Policy prediction ended with events '<span class="error">[
 <rasa.shared.core.events.defineprevuserutteredfeaturization object at 0x1b1b60460>
  ]
 </rasa.shared.core.events.defineprevuserutteredfeaturization></span>'.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Action 'action_unlikely_intent' ended with events '[]'.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Current slot values:
 my_slot: None
 my_form_trigger: None
 requested_slot: None
 session_started_metadata: None
 2022-03-31 12:22:31 DEBUG rasa.engine.runner.dask - Running graph with inputs: 

 {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x1b1aed160>} 

, targets: <span class="error">['select_prediction']</span> and ExecutionContext(model_id='11ec92063cb6476b839b182307917603', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'domain_provider' running 'DomainProvider.provide_inference'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.memoization - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 <span class="error">[state 3]</span> user intent: greet | previous action name: action_listen
 2022-03-31 12:22:31 DEBUG rasa.core.policies.memoization - There is no memorised next action
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.rule_policy - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 <span class="error">[state 3]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 4]</span> user intent: greet | previous action name: action_unlikely_intent
 2022-03-31 12:22:31 DEBUG rasa.core.policies.rule_policy - There is no applicable rule.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.ted_policy - TED predicted 'utter_greet' based on user intent.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.unexpected_intent_policy - Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.ensemble - Predicted next action using RulePolicy.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Predicted next action 'action_default_fallback' with confidence 0.30.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Policy prediction ended with events '[]'.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Action 'action_default_fallback' ended with events '<span class="error">[
 <rasa.shared.core.events.userutterancereverted object at 0x1b1addfa0>
  ]
 </rasa.shared.core.events.userutterancereverted></span>'.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Current slot values:
 my_slot: None
 my_form_trigger: None
 requested_slot: None
 session_started_metadata: None
 2022-03-31 12:22:31 DEBUG rasa.engine.runner.dask - Running graph with inputs: 

 {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x1b1aed160>} 

, targets: <span class="error">['select_prediction']</span> and ExecutionContext(model_id='11ec92063cb6476b839b182307917603', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'domain_provider' running 'DomainProvider.provide_inference'.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.memoization - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 2022-03-31 12:22:31 DEBUG rasa.core.policies.memoization - There is a memorised next action 'action_listen'
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.rule_policy - Current tracker state:
 <span class="error">[state 1]</span> user intent: greet | previous action name: action_listen
 <span class="error">[state 2]</span> user intent: greet | previous action name: utter_greet
 2022-03-31 12:22:31 DEBUG rasa.core.policies.rule_policy - There is no applicable rule.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.ted_policy - TED predicted 'action_listen' based on user intent.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.unexpected_intent_policy - Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
 2022-03-31 12:22:31 DEBUG rasa.engine.graph - Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
 2022-03-31 12:22:31 DEBUG rasa.core.policies.ensemble - Predicted next action using MemoizationPolicy.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Predicted next action 'action_listen' with confidence 1.00.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Policy prediction ended with events '[]'.
 2022-03-31 12:22:31 DEBUG rasa.core.processor - Action 'action_listen' ended with events '[]'.
 2022-03-31 12:22:31 DEBUG rasa.core.lock_store - Deleted lock for conversation '61f6d7f4fa2f44e7834b71cf72e8d2b7'.

**Comment by Hendler** [comment]
TL;DR

The solution was to add two `FollowupAction` events and stories.

I’m probably not understanding Rasa very well, but as a new user I’d like to see an easier way to do do this if there was an event that **could trigger a rule or intent directly**. It appears that a couple actions being triggered allows a story to be used, which seems like a hack compared to triggering an intent or rule directly.

```
        slot_events.append(FollowupAction(name="utter_hello_form"))
        slot_events.append(FollowupAction(name="my_form"))
```
